lib/bolt/config/transport.rb

```diff
@@ -0,0 +1,133 @@
+# frozen_string_literal: true
+
+require 'bolt/error'
+require 'bolt/util'
+
+module Bolt
+  class Config
+    class Transport
+      attr_reader :config
+
+      def initialize(data = {}, boltdir = nil)
+        assert_hash(data)
+        @config = Bolt::Util.deep_merge(defaults, filter(data))
+        @boltdir = boltdir
+        validate
+      end
+
+      def [](key)
+        @config[key]
+      end
+
+      def to_h
+        @config
+      end
+
+      def fetch(*args)
```

Comment:
This is used by the docker and sudoable transports, which currently have `target.options.fetch`. I can either update `Bolt::Inventory::Target#options` to return the transport config hash, or update the transports to `target.options.to_h.fetch`. Not sure which would be preferable, so I put this in for now.